doc/architecture/decisions/0013-p2p-ipc-abstraction.md

```diff
@@ -0,0 +1,32 @@
+# 13. P2P - IPC Abstraction
+
+Date: 2018-07-19
+
+## Status
+
+Proposed
+
+## Context
+
+It's frustrating to admit, but I'm still pretty slow at writing rust code. Oh, I can hack something together quickly (gogo `unwrap()/clone()`), but when I sit down to write efficient production code, I get bogged down in cyclic lifetime reference issues or any number of other arcane rustisisms.
+
+In this prototyping / proof-of-concept phase, I don't want to take the time to write efficient rust code, but I don't want to leave us with a bunch of unmaintainable inefficient spaghetti if we *do* decide to go forward with the prototype.
+
+## Decision
+
+Abstract the p2p library at the process level.
+
+- Host a [ZeroMQ](http://zeromq.org/) REP socket for the p2p process. This process can be in any language that supports zmq, and using any of the transports it supports. (Likely start with unix domain sockets for their high throughput).
```

Comment:
what are the consequences for Windows users?  Will they have to install a separate component?

Reply:
zeromq in general works on windows, and the rust specific zmq crate does too:
- https://github.com/erickt/rust-zmq/pull/124
- https://github.com/erickt/rust-zmq/issues/125